### PR TITLE
Add agreements

### PR DESCRIPTION
**Pull request by huwshimi**
Add the methods to add user agreement to terms.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- set up the gui to talk to GUIMAAS
- in your browser console run `app.terms.getAgreements((error, data) => {console.log(error, data)})`
- you should get `null, null`
- now run `app.terms.addAgreement([{name: 'lorem-ipsum', revision: 4}], (error, data) => {console.log(error, data)})`
- you should get back a response with the details of the term agreed to
- run `app.terms.getAgreements((error, data) => {console.log(error, data)})`
- you should get an array including the term you just agreed to

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5318/
Test PASSed.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js [hunk 1]
```diff
@@ -79,10 +79,61 @@ var module = module;
...
+      @param terms {Array} A list of terms to agree to. Each term should be an
+        object with the following parameters:
+        - name {String} The terms name.
+        - revision {Int} The terms revision.
+      @param callback {Function} A callback to handle errors or accept the
+        data from the request. Must accept an error message or null as its
+        first parameter and an authorization object as its second.
+    */
+    addAgreement: function(terms, callback) {
+      var self = this;
```
Comment (hatched) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js]: If you use arrow fn's for 112 and 117 then the context will be the same context as you'd be saving here.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js [hunk 2]
```diff
@@ -79,10 +79,61 @@ var module = module;
...
+          callback(error, null);
+          return;
+        }
+        var terms = response.agreements.map(function(term) {
+          return self._formatTerm(term);
+        });
+        callback(null, terms);
+      }
+      var url = this.url + '/agreement';
+      var payload = terms.map(function(term) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js]: You should make this smaller by doing the following if you like :)

```JavaScript
payload = terms.map(term => { termname: term.name, termrevision: term.revision});
```

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js]: I had thought we were trying not to use es6 in jujulib, but I could be remembering wrong.

Comment (hatched) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js]: Ahh I believe you're right

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js [hunk 3]
```diff
@@ -79,10 +79,61 @@ var module = module;
...
+        return {
+          termname: term.name,
+          termrevision: term.revision
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js]: Should these names be camelcase?

**Comment by anthonydillon** [comment]
Good work Huw. :+1: with a question.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js (hunk 3 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/terms.js]: According to the code they're just lowercase, one word: https://github.com/CanonicalLtd/terms-service/blob/master/params/params.go#L59

**Comment by hatched** [comment]
:+1: Thanks for this @huwshimi

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**